src/cswinrt/code_writers.h

```diff
@@ -9037,11 +9074,33 @@ private static unsafe int Do_Abi_Invoke%
             type.TypeName(),
             type_params,
             [&](writer& w) {
-                if (type_params.empty()) return;
-                w.write(R"(
-public static Guid PIID = GuidGenerator.CreateIID(typeof(%));)",
-                    type_name
-                );
+                if (!type_params.empty())
+                {
+                    // Generating both PIID and IID for backcompat consistency and to have a common property to rely on
+                    // similar to what we do in the manual projections.
+                    w.write(R"(
```

Comment:
Not sure I get it, why do we need consistency if this is only for new projections? And why do we need two in the first place rather than just IID even for the generic case? 🤔

Reply:
PIID was pre-existing, so if there was a previously generated projection that had a dependency on the old version of this, removing it can break it.  At the same time, I have been generalizing across all the interfaces having an IID property that they can call to get the GUID rather than checking if it is a generic or not first and switching between PIID and IID.  Due to that, I added both here.